### DISCUSSION TITLE
How can I run my own private scripts on the top of the Packer creation process?

### DISCUSSION THREAD

**Discussion by zioalex** [actions/virtual-environments#4905] [Image configuration]
Hi dear community,
I am evaluating the use of the virtual-environments for my company.

I have the challenge that I need to run some specific and private scripts into the image on the top of the standard process.
Is there any way to plugin my configuration?

**Comment by mmuffins** [comment]
We had the same requirement and came to the conclusion that updating the packer script locally is more trouble than it's worth since you constantly need to merge changes from the upstream repository. Instead, we build the image as-is and created it a new, local, packer template that uses the previously built image as base and applies the additional customizations on top of it.
